rental.rb

```diff
@@ -0,0 +1,22 @@
+class Rental
+  attr_reader :person, :book, :date
+
+  def initialize(date, person, book)
+    @date = date
+    @book = book
+    @person = person
+
+    book.rentals.push(self)
+    person.rentals.push(self)
+  end
+
+  def book=(book)
+    @book = book
+    book.rentals.push(self) unless book.rentals.include?(self)
+  end
+
+  def person=(person)
+    @person = person
+    person.renatls.push(self) unless person.rentals.include?(self)
+  end
```

Comment:
- [ ] What hasn't been realized here is that these two methods are a redundancy. When you look at the two lines below 👇🏼:

  ```ruby
  book.rentals.push(self)
  person.rentals.push(self)
  ```
  
  They already tell us that the initialize block has already taken care of the `belongs to` association between a rental and book and between a rental and a person.
  
  Our main aim here is to have rental add itself to `book.rentals` and `person.rentals.` Therefore these two methods are really not needed because they are repeating what the initialize block is already doing. @Gedewon, please refactor these two methods out of this class and we'll be good to go.

Reply:
@Lemechk1  Thank you for the review !!! 
yes I am assuming that joint table exist without person and book  and handle when we have rental without person and book for safe keeping , but it's clear that it should't in our case !! Thank you I will make the nessesary changes.